### PR TITLE
Potential Cross Site Scripting ( XSS )

### PR DESCRIPTION
**Pull request by shashank-elastic**
## Issues
#2820

## Summary
- Testing Details Updated in main issue.

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

**Review by DefSecSentinel — APPROVED** [review]
LGTM

File: rules/apm/initial_access_cross_site_scripting.toml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+rule_id = "4aa58ac6-4dc0-4d18-b713-f58bf8bd015c"
+severity = "medium"
+tags = ["Data Source: APM", "Use Case: Threat Detection", "Tactic: Initial Access"]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+any where processor.name == "transaction" and
+url.fragment : ("*iframe*", "*prompt*", "*script*", "*svg*", "*onerror=*", "*javascript*alert*", "*eval*", "*onclick*",
+"*document.cookie*", "*document.domain*","*onresize=*","*onload=*","*onmouseover=*", "*${alert*")
```
Comment (Aegrah) [inline rules/apm/initial_access_cross_site_scripting.toml:31]:
```suggestion
"*document.cookie*", "*document.domain*","*onresize=*","*onload=*","*onmouseover=*")
```

**Comment by Aegrah** [comment]
Is there a reason you chose to just focus on url fragments and not on actual web content in POST requests for example? 

XSS within a URL is usually reflected XSS, because it is not actually stored on the web page. As this makes it more difficult to steal cookies or perform other attacks without additional attack vectors, this is in most cases not that harmful. Stored XSS however is a serious issue, but this will usually not be reflected in a web page URL, but more commonly in stored fields.

File: rules/apm/initial_access_cross_site_scripting.toml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+min_stack_comments = "New fields added: required_fields, related_integrations, setup"
+min_stack_version = "8.3.0"
+updated_date = "2023/07/12"
+
+[rule]
+author = ["Elastic"]
+description = """
+Cross-Site Scripting (XSS) attacks are a type in which malicious scripts are injected into trusted websites.
+In XSS attacks, an attacker uses a benign web application to send malicious code, generally in the form of a
+browser-side script.This detection rule identifies the potential malicious executions of such browser-side scripts.
```
Comment (w0rk3r) [inline rules/apm/initial_access_cross_site_scripting.toml:14]:
```suggestion
Cross-Site Scripting (XSS) is a type of attack in which malicious scripts are injected into trusted websites.
In XSS attacks, an attacker uses a benign web application to send malicious code, generally in the form of a
browser-side script. This detection rule identifies the potential malicious executions of such browser-side scripts.
```

File: rules/apm/initial_access_cross_site_scripting.toml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+"""
+index = ["apm-*-transaction*", "traces-apm*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Potential Cross Site Scripting ( XSS )"
```
Comment (w0rk3r) [inline rules/apm/initial_access_cross_site_scripting.toml:19]:
```suggestion
name = "Potential Cross Site Scripting (XSS)"
```

File: rules/apm/initial_access_cross_site_scripting.toml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+references = ["https://github.com/payloadbox/xss-payload-list"]
+risk_score = 47
+rule_id = "4aa58ac6-4dc0-4d18-b713-f58bf8bd015c"
+severity = "medium"
+tags = ["Data Source: APM", "Use Case: Threat Detection", "Tactic: Initial Access"]
+timestamp_override = "event.ingested"
```
Comment (w0rk3r) [inline rules/apm/initial_access_cross_site_scripting.toml:25]:
```suggestion
risk_score = 21
rule_id = "4aa58ac6-4dc0-4d18-b713-f58bf8bd015c"
severity = "low"
tags = ["Data Source: APM", "Use Case: Threat Detection", "Tactic: Initial Access"]
timestamp_override = "event.ingested"
building_block_type = "default"
```

File: rules/apm/initial_access_cross_site_scripting.toml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+updated_date = "2023/07/12"
+
+[rule]
+author = ["Elastic"]
+description = """
+Cross-Site Scripting (XSS) attacks are a type in which malicious scripts are injected into trusted websites.
+In XSS attacks, an attacker uses a benign web application to send malicious code, generally in the form of a
+browser-side script.This detection rule identifies the potential malicious executions of such browser-side scripts.
+"""
+index = ["apm-*-transaction*", "traces-apm*"]
```
Comment (w0rk3r) [inline rules/apm/initial_access_cross_site_scripting.toml:16]:
```suggestion
"""
from = "now-119m"
interval = "60m"
index = ["apm-*-transaction*", "traces-apm*"]
```

File: rules/apm/initial_access_cross_site_scripting.toml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+risk_score = 47
+rule_id = "4aa58ac6-4dc0-4d18-b713-f58bf8bd015c"
+severity = "medium"
+tags = ["Data Source: APM", "Use Case: Threat Detection", "Tactic: Initial Access"]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+any where processor.name == "transaction" and
+url.fragment : ("*iframe*", "*prompt*", "*script*", "*svg*", "*onerror=*", "*javascript*alert*", "*eval*", "*onclick*",
```
Comment (w0rk3r) [inline rules/apm/initial_access_cross_site_scripting.toml:30]: As is, `*script*` for example, will match unrelated stuff such as `description, subscription, etc`:
image

Comment (w0rk3r) [reply rules/apm/initial_access_cross_site_scripting.toml:30]: I think there is a lot of room in matching page names too:
image

Comment (w0rk3r) [reply rules/apm/initial_access_cross_site_scripting.toml:30]: `*eval*` would match `retrieval` and potentially other unrelated stuff

**Review by w0rk3r — CHANGES_REQUESTED** [review]
As this is a new subject which we haven't tested which a lot of production data, I think we should convert and move this one to BBR to collect more data and have an idea on how noisy the rule is as we don't have a deprecation mechanism.

We should also evaluate using new_terms instead, to avoid flooding the alerts interface with repeated activity.

**Comment by shashank-elastic** [comment]
@w0rk3r Agreed on the feedback this could move to a BBR mode to evaluate its potentiality of being an actual detection rule. Accepted the suggestions and moved them to the BBR folder. Although it's an APM integration unit tests mandate it to be in the rules_building_block folder.

@Aegrah With APM logs s we could achieve reflected Xss detection at this point and the rule was primarily focused on those. Stored XSS I would see if those can be detected through APM logs.

File: rules/apm/initial_access_cross_site_scripting.toml (hunk 6 above)
Comment (w0rk3r) [reply rules/apm/initial_access_cross_site_scripting.toml]: @shashank-elastic I think we can work on eliminating these partial matches before merging this. Consider combining them with other pieces of the payload.

**Review by Aegrah — APPROVED** [review]

File: rules/apm/initial_access_cross_site_scripting.toml (hunk 6 above)
Comment (shashank-elastic) [reply rules/apm/initial_access_cross_site_scripting.toml]: Have added move htm constructs like '<' , '=' in wildcard expresssions to mimic the payload.
With eval generally landing to a parenthesis payload those are additionally added aswell.
